### PR TITLE
samples: nrf_desktop: Fix swift pair build dependencies

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-416

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Comment by pdunaj** [comment]
rebased
tested on PCA20041

**Comment by pdunaj** [comment]
Hi @kapi-no , could you also have a look as @Qbicz is on holiday?

File: samples/nrf_desktop/src/modules/ble_adv.c [hunk 1]
```diff
@@ -136,25 +151,64 @@ static int ble_adv_start(void)
...
+{
+	int err = bt_le_adv_update_data(ad, (ARRAY_SIZE(ad) - SWIFT_PAIR_SECTION_SIZE),
+					sd, ARRAY_SIZE(sd));
+
+	if (err) {
+		LOG_ERR("Cannot modify advertising data (err %d)", err);
+	} else {
+		LOG_INF("Vendor section removed");
+
+		if (IS_ENABLED(CONFIG_DESKTOP_BLE_FAST_ADV)) {
```
Comment (kapi-no) [inline samples/nrf_desktop/src/modules/ble_adv.c]: This if statement seems redundant.

This function can only be called if CONFIG_DESKTOP_BLE_SWIFT_PAIR is enabled
&
CONFIG_DESKTOP_BLE_SWIFT_PAIR depends on CONFIG_DESKTOP_BLE_FAST_ADV

Comment (pdunaj) [reply samples/nrf_desktop/src/modules/ble_adv.c]: this is true; I have added it for symmetry as the dependency is hidden in the Kconfig and may not be obvious; If you want I can remove it so please tell

Comment (kapi-no) [reply samples/nrf_desktop/src/modules/ble_adv.c]: you can leave it like it is now

Comment (pdunaj) [reply samples/nrf_desktop/src/modules/ble_adv.c]: thx

**Comment by pdunaj** [comment]
Hi @pizi-nordic , could you merge?

**Pull request merged**